Packs/TOPdesk/Integrations/TOPdesk/TOPdesk.py

```diff
@@ -1272,6 +1295,226 @@ def fetch_incidents(client: Client,
+        update_args = {
+            'id': ticket_id
+        }
+        for key in parsed_args.delta:
+            update_args[key] = parsed_args.delta[key]
+
+        demisto.debug(f'SZU update_args=[{update_args}]')
+        client.update_incident(update_args)
+
+    entries = parsed_args.entries
+    if entries:
+        demisto.debug(f'New entries {entries}')
+
+        for entry in entries:
+            demisto.debug(f'Sending entry {entry.get("id")}, type: {entry.get("type")}')
+            # Mirroring files as entries
+            if entry.get('type') == 3:
+                path_res = demisto.getFilePath(entry.get('id'))
+                full_file_name = path_res.get('name')
+                file_name, file_extension = os.path.splitext(full_file_name)
+                if not file_extension:
+                    file_extension = ''
+                client.attachment_upload(incident_id=ticket_id, incident_number=None, file_entry=entry.get('id'),
+                                         file_name=file_name + '_mirrored_from_xsoar' + file_extension,
+                                         invisible_for_caller=False,
+                                         file_description=f"Upload from xsoar: {file_name}.{file_extension}")
+            else:
+                # Mirroring comment and work notes as entries
+                xargs = {
+                    'id': '',
+                    'action': '',
+                    'action_invisible_for_caller': False,
+                }
+                tags = entry.get('tags', [])
+                if params.get('work_notes_tag') in tags:
+                    xargs['action_invisible_for_caller'] = True
+                elif params.get('comments_tag') in tags:
+                    xargs['action_invisible_for_caller'] = False
+                # Sometimes user is an empty str, not None, therefore nothing is displayed
+                user = entry.get('user', 'dbot')
+                if (user):
+                    duser = demisto.findUser(username=user)
+                    name = duser['name']
+                else:
+                    name = 'Xsoar dbot'
+
+                text = f"<i><u>Update from {name}:</u></i><br><br>{str(entry.get('contents', ''))}" \
+                    + "<br><br><i>Mirrored from Cortex XSOAR</i>"
+                # client.add_comment(ticket_id, ticket_type, key, text)
+
+                xargs['id'] = ticket_id
+                xargs['action'] = text
+                client.update_incident(xargs)
+
+    return ticket_id
+
+
+def get_mapping_fields_command(client: Client) -> GetMappingFieldsResponse:
+    """
+    Returns the list of fields for an incident type.
+    Args:
+        client: Xsoar client to use
+
+    returns: Dictionairy with keys as field names
+
+    """
+
+    incident_type_scheme = SchemeTypeMapping(type_name=client.ticket_type)
+    demisto.debug(f'Collecting incident mapping for incident type - "{client.ticket_type}"')
```

Comment:
I see that the only `ticket_type` is `'incident'`, so it can be changed here. The `type_name` here needs to be the name that is in the `yml` of the incident type, so it can't be just `incident`.
```suggestion
    incident_type_scheme = SchemeTypeMapping(type_name='TOPdesk Incident')
    demisto.debug(f'Collecting incident mapping for incident type - TOPdesk Incident')
```

Reply:
I copied this from the ServiceNowV2 integration.
For now we only support the incident module in TOPdesk, but maybe later the change.
But that module is much more complex to support.

Is it ok to let is like it was?

Reply:
No, as it has a different behavior than wanted. The changes is only in XSOAR- so the mapping scheme will be related to the right incident type.